gum/backend-libdwarf/gumsymbolutil-libdwarf.c

```diff
@@ -344,6 +356,58 @@ gum_find_function (const gchar * name)
   return address;
 }
 
+static gboolean
+gum_find_nearest_symbol_by_address (const gpointer address,
+                                    GumNearestSymbolDetails * nearest)
+{
+  GHashTable * table;
+  gchar * name;
+  GList * keys;
+  GList * current;
+  gpointer current_address;
+  gpointer closest_address = NULL;
+
+  table = gum_get_address_symbols ();
+  name = g_hash_table_lookup (table, address);
+
+  if (name != NULL)
+  {
+    nearest->name = name;
+    nearest->address = address;
+    return TRUE;
+  }
+
+  keys = g_hash_table_get_keys (table);
```

Comment:
Perhaps doing `closest_address = NULL;` right before `keys =` would make it clearer what we're doing here.